src/Agent.Worker/TestResults/JunitResultReader.cs

```diff
@@ -0,0 +1,311 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Xml;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public class JUnitResultReader : AgentService, IResultReader
+    {
+        public Type ExtensionType => typeof(IResultReader);
+
+        public JUnitResultReader()
+        {
+            AddResultsFileToRunLevelAttachments = true;
```

Comment:
you don't really need the constructor. since you don't change it to anything.
public bool AddResultsFileToRunLevelAttachments {get {return true}}; already enought,